src/shared/Microsoft.Git.CredentialManager.Tests/HostProviderRegistryTests.cs

```diff
@@ -57,5 +83,140 @@ public void HostProviderRegistry_MultipleValidProviders_ReturnsFirstRegistered()
+        }
+
+        [Fact]
+        public void HostProviderRegistry_GetProvider_UnknownProviderSpecified_ReturnsFirstSupportedProvider()
+        {
+            var context = new TestCommandContext
+            {
+                Settings = {ProviderOverride = "provider42"}
+            };
+            var registry = new HostProviderRegistry(context);
+            var input = new InputArguments(new Dictionary<string, string>());
+
+            var provider1Mock = new Mock<IHostProvider>();
+            var provider2Mock = new Mock<IHostProvider>();
+            var provider3Mock = new Mock<IHostProvider>();
+            provider1Mock.Setup(x => x.Id).Returns("provider1");
+            provider1Mock.Setup(x => x.IsSupported(It.IsAny<InputArguments>())).Returns(false);
+            provider2Mock.Setup(x => x.Id).Returns("provider2");
+            provider2Mock.Setup(x => x.IsSupported(It.IsAny<InputArguments>())).Returns(true);
+            provider3Mock.Setup(x => x.Id).Returns("provider3");
+            provider3Mock.Setup(x => x.IsSupported(It.IsAny<InputArguments>())).Returns(false);
+
+            registry.Register(provider1Mock.Object, provider2Mock.Object, provider3Mock.Object);
+
+            IHostProvider result = registry.GetProvider(input);
+
+            Assert.Same(provider2Mock.Object, result);
+        }
+
+        [Fact]
+        public void HostProviderRegistry_GetProvider_LegacyAuthoritySpecified_ReturnsProvider()
+        {
+            var context = new TestCommandContext
+            {
+                Settings = {LegacyAuthorityOverride = "authorityB"}
+            };
+            var registry = new HostProviderRegistry(context);
+            var input = new InputArguments(new Dictionary<string, string>());
+
+            var provider1Mock = new Mock<IHostProvider>();
+            var provider2Mock = new Mock<IHostProvider>();
+            var provider3Mock = new Mock<IHostProvider>();
+            provider1Mock.Setup(x => x.SupportedAuthorityIds).Returns(new[]{"authorityA"});
+            provider1Mock.Setup(x => x.IsSupported(It.IsAny<InputArguments>())).Returns(false);
+            provider2Mock.Setup(x => x.SupportedAuthorityIds).Returns(new[]{"authorityB", "authorityC"});
+            provider2Mock.Setup(x => x.IsSupported(It.IsAny<InputArguments>())).Returns(false);
+            provider3Mock.Setup(x => x.SupportedAuthorityIds).Returns(new[]{"authorityD"});
+            provider3Mock.Setup(x => x.IsSupported(It.IsAny<InputArguments>())).Returns(false);
+
+            registry.Register(provider1Mock.Object, provider2Mock.Object, provider3Mock.Object);
+
+            IHostProvider result = registry.GetProvider(input);
+
+            Assert.Same(provider2Mock.Object, result);
+        }
+
+        [Fact]
+        public void HostProviderRegistry_GetProvider_AutoLegacyAuthoritySpecified_ReturnsFirstSupportedProvider()
+        {
+            var context = new TestCommandContext
```

Comment:
Conceptually it would help me if we used actual names (azure, github, etc..) instead of provider1/provider2

Reply:
Let me know your thoughts on this one.  Now that I know the meaning really well I certainly won't forget it, but I do think actual names are quicker to digest.